features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/NotificationSettingsPresenter.kt

```diff
@@ -68,6 +76,61 @@ class NotificationSettingsPresenter @Inject constructor(
             observeNotificationSettings(matrixSettings)
         }
 
+        // List of PushProvider -> Distributor
+        val distributors = remember {
+            pushService.getAvailablePushProviders()
+                .flatMap { pushProvider ->
+                    pushProvider.getDistributors().map { distributor ->
+                        pushProvider to distributor
+                    }
+                }
+        }
+        // List of Distributor names
+        val distributorNames = remember {
+            distributors.map { it.second.name }
+        }
+
+        var currentDistributorName by remember { mutableStateOf<AsyncAction<String>>(AsyncAction.Uninitialized) }
```

Comment:
Should this be `AsyncData` instead?

Reply:
Could be but `currentDistributorName` is also updated when the user change the distributor, so it's also an AsyncAction... The boundaries between the 2 sealed interface is thin.

Reply:
I mean, having an 'action' called `currentDistributorName` is kind of weird, that's why I thought `AsyncData` fits it better. More so if we're going to use the `LaunchedEffect` to fetch it.

Reply:
OK: 9a24e7d4afb0d1aba94808f4327949baccc68883